### ISSUE TITLE
Autoclose breaks <C-h> mapping

### ISSUE THREAD

**Issue by NielsCodes** [m4xshen/autoclose.nvim#57]
I have the following keymaps in my config:

```lua
vim.keymap.set("i", "<C-h>", "<Left>", {})
vim.keymap.set("i", "<C-l>", "<Right>", {})
vim.keymap.set("i", "<C-j>", "<Down>", {})
vim.keymap.set("i", "<C-k>", "<Up>", {})
```

This is so I can navigate through a line with Control + hjkl while in insert mode. However, Ctrl + h would not move the cursor, but instead it would perform a backspace. 

This lead me down a rabbit hole of <C-h> mapping issues like [this thread](https://github.com/neovim/neovim/issues/2048) and [this one](https://github.com/neovim/neovim/issues/2294).

After a lot of trial and just as much error, I came to the conclusion that it was actually this plugin that was breaking my keymap.

**Comment by NielsCodes** [comment]
And for anyone wondering how I fixed it: 
I switched to [nvim-autopairs](https://github.com/windwp/nvim-autopairs) instead.

**Comment by telemachus** [comment]
`<C-h>` has a built-in function [in insert mode](https://neovim.io/doc/user/insert.html#i_CTRL-H) and [in command mode](https://neovim.io/doc/user/cmdline.html#c_CTRL-H): it deletes the previous character. This plugin assumes that as a default, and [preserves that behavior](https://github.com/m4xshen/autoclose.nvim/blob/main/lua/autoclose.lua#L104-L105). That makes sense, from one point of view, since it is the default behavior of `C-h` in insert and command mode.

On the other hand, I see from your links that `C-h` as `<Left>` in insert mode is a reasonably popular custom mapping. This is a general problem with {neo,}vim plugins: they may create mappings that mess with a user's mappings. I'm not sure what the right response is here. One thought is that the plugin could offer a configuration option to opt-out of remapping `C-h` in one or more modes. We'll see what @m4xshen thinks though.

**Comment by NielsCodes** [comment]
@telemachus I see your point about respecting defaults and I know that `<C-h>` is a special case, but I personally don't agree that it makes sense for a plugin to assume this (or any other) default mapping is actually in use. 

And maybe for a little context as to why this `<Left>` mapping is so popular: 
Using Tmux + Neovim and something like [vim-tmux-navigator](https://github.com/christoomey/vim-tmux-navigator), you can easily move between all your terminal panes and Neovim windows with CTRL + hjkl. Being able to use those same keymaps for navigation within insert mode is great, because navigation works the same way regardless of the level at which you're navigating. It's 100% muscle memory at this point, so not being able to use that keymap is a deal-breaker for me.

**Comment by telemachus** [comment]
> I personally don't agree that it makes sense for a plugin to assume this (or any other) default mapping is actually in use

Fair enough (though probably awfully complicated in practice). In this case, I suspect that if the mapping added [`unique`](https://neovim.io/doc/user/map.html#_1.2-special-arguments), that might fix the problem. I maintain a fork of this repo (with some tweaks), and I'll test it out there. If things work out, I'll submit a PR here and ping you. Also, as I said originally, the plugin's creator may have other thoughts too.

**Comment by telemachus** [comment]
@NielsCodes If you have time, try [my fork](https://github.com/telemachus/autoclose.nvim/tree/mapcheck) using the "mapcheck" branch. It wasn't [a lot of code](https://github.com/telemachus/autoclose.nvim/commit/e915903cfa2c09e2e99d8cf80842f3edf9143215), but I need to test it further and think about it more before I make a pull request.

**Comment by NielsCodes** [comment]
@telemachus sorry for the delay. I tried your fork and it still does not seem to work. I appreciate you looking into this, but I guess I'm pretty happy with nvim-autopairs now. So if it's a pain to implement, I don't want to waste your time on this either.

**Comment by telemachus** [comment]
@NielsCodes No problem. I have a few other ideas, but if you found something you're happy with, I'm glad.
